src/OrchardCore.Modules/OrchardCore.Contents/Controllers/AdminController.cs

```diff
@@ -234,63 +233,66 @@ public ActionResult ListFilterPOST(ListContentsViewModel model)
-        //                    }
-
-        //                    _contentManager.Publish(item);
-        //                }
-        //                Services.Notifier.Information(T("Content successfully published."));
-        //                break;
-        //            case ContentsBulkAction.Unpublish:
-        //                foreach (var item in checkedContentItems)
-        //                {
-        //                    if (!await _authorizationService.Authorize(User, Permissions.PublishContent, item, T("Couldn't unpublish selected content.")))
-        //                    {
-        //                        _transactionManager.Cancel();
-        //                        return Unauthorized();
-        //                    }
-
-        //                    _contentManager.Unpublish(item);
-        //                }
-        //                Services.Notifier.Information(T("Content successfully unpublished."));
-        //                break;
-        //            case ContentsBulkAction.Remove:
-        //                foreach (var item in checkedContentItems)
-        //                {
-        //                    if (!await _authorizationService.Authorize(User, Permissions.DeleteContent, item, T("Couldn't remove selected content.")))
-        //                    {
-        //                        _transactionManager.Cancel();
-        //                        return Unauthorized();
-        //                    }
-
-        //                    _contentManager.Remove(item);
-        //                }
-        //                Services.Notifier.Information(T("Content successfully removed."));
-        //                break;
-        //            default:
-        //                throw new ArgumentOutOfRangeException();
-        //        }
-        //    }
-
-        //    return this.RedirectLocal(returnUrl, () => RedirectToAction("List"));
-        //}
+        [HttpPost, ActionName("List")]
+        [FormValueRequired("submit.BulkAction")]
+        public async Task<ActionResult> ListPOST(ContentOptions options, IEnumerable<int> itemIds)
+        {
+            if (itemIds.Count() > 0)
+            {
+                var checkedContentItems = await _session.Query<ContentItem, ContentItemIndex>().Where(x => x.DocumentId.IsIn(itemIds) && x.Latest).ListAsync();
+                switch (options.BulkAction)
+                {
+                    case ContentsBulkAction.None:
+                        break;
+                    case ContentsBulkAction.PublishNow:
+                        foreach (var item in checkedContentItems)
+                        {
+                            if (!await _authorizationService.AuthorizeAsync(User, Permissions.PublishContent, item))
+                            {
+                                //_notifier.Warning(T["Couldn't publish selected content."]);
+                                //_transactionManager.Cancel();
```

Comment:
I kept those 2 lines in comment. The notifier message in 01 was passed on the authorizationService ; we don't have this in OC. The transactionManager I can't say if it's required now with YesSQL.

Reply:
In this case call `ISession.CancelAsync()` because it would publish the previous items. And I assume it's better to cancel everything than just stop in the middle. But do whatever a single "Publish" action does right now.

The other solution would be to first check all the permissions, then if all is good start publishing them, or return an error message.

Same for the other calls to Authorize